ios/FluentUI/People Picker/ContactView.swift

```diff
@@ -0,0 +1,189 @@
+    ///
+    /// - Parameters:
+    ///   - title: String that will be the text of the top label
+    ///   - subtitle: String that will be the text of the bottom label
+    @objc public convenience init(title: String, subtitle: String) {
+        self.init(title: title, subtitle: subtitle, identifier: nil)
+    }
+
+    /// Initializes the contact view by creating an avatar view with an identifier
+    ///
+    /// - Parameters:
+    ///   - identifier: String that will be used to identify the contact (e.g. email, phone number, first name)
+    @objc public convenience init(identifier: String) {
+        self.init(title: nil, subtitle: nil, identifier: identifier)
+    }
+
+    private init(title: String?, subtitle: String?, identifier: String?) {
+        avatarView = AvatarView(avatarSize: .extraExtraLarge, withBorder: false, style: .circle)
+        super.init(frame: .zero)
+
+        if let title = title, let subtitle = subtitle {
+            subtitleLabel = UILabel()
+            setupAvatarView(with: title, and: subtitle, or: nil)
+            setupTitleLabel(using: title)
+            setupSubtitleLabel(using: subtitle)
+        } else if let identifier = identifier {
+            setupAvatarView(with: nil, and: nil, or: identifier)
+            setupTitleLabel(using: identifier)
+        }
+
+        backgroundColor = Colors.surfacePrimary
+        setupLayout()
+    }
+
+    public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    private func setupAvatarView(with firstName: String?, and lastName: String?, or identifier: String?) {
+        if let firstName = firstName, let lastName = lastName {
+            let fullName = firstName + " " + lastName
+            avatarView.setup(primaryText: fullName, secondaryText: identifier, image: avatarImage)
+        } else {
+            avatarView.setup(primaryText: identifier, secondaryText: "", image: avatarImage)
+        }
+    }
+
+    private func setupLayout() {
+        var constraints = [NSLayoutConstraint]()
+        constraints.append(widthAnchor.constraint(equalToConstant: Constants.contactWidth))
+        constraints.append(contentsOf: avatarLayoutConstraints())
+
+        if let titleLabel = titleLabel, let subtitleLabel = subtitleLabel {
+            constraints.append(contentsOf: titleLabelLayoutConstraints())
+            constraints.append(contentsOf: subtitleLabelLayoutConstraints())
+            addSubview(titleLabel)
+            addSubview(subtitleLabel)
+        } else if let titleLabel = titleLabel {
+            constraints.append(contentsOf: identifierLayoutConstraints())
+            addSubview(titleLabel)
+        }
+
+        addSubview(avatarView)
+
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    private func avatarLayoutConstraints() -> [NSLayoutConstraint] {
+        avatarView.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
>avatarView.translatesAutoresizingMaskIntoConstraints = false [](start = 8, length = 60)

this belongs next to the line that adds it as a subview. This is only necessary before the constraints are activated. Same in the other constraint method

Reply:
Why does the line belong next to the line that adds it as a subview? From my code, `avatarLayoutConstraints()` is always called before the constraints are activated. Is it because we only want to set the `translatesAutoresizingMaskIntoConstraints` property a single time and not each time `avatarLayoutConstraints()` is called?

Reply:
that is one of the reasons. But also because this method's name indicates it's a getter and doesn't indicate it will perform an action like this. Also, translatesAutoresizingMaskIntoConstraints is important before activating constraints so it makes sense that code is in the same place